File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;
        int[] answer = {0, 0};
        
        while (left < right) {
            if (numbers[left] + numbers[right] > target) {
                right--;
            } else if (numbers[left] + numbers[right] < target) {
                left++;
            } else {
                answer[0] = left + 1;
                answer[1] = right + 1;
                return answer;
            }
        }
        
        return answer;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var left = 0;
        var right = numbers.size - 1;
        val answer = intArrayOf(0, 0)

        while (left < right) {
            val r = numbers[right]
            val l = numbers[left]

            if (target < r + l) {
                right--
            } else if (target > r + l) {
                left++
            } else {
                answer[0] = left + 1
                answer[1] = right + 1
                return answer
            }
        }

        return answer
    }
}